gallery/gallery-library/src/main/java/org/silverpeas/components/gallery/MediaUtil.java

```diff
@@ -369,288 +339,362 @@ public static void setMetaData(final FileHandler fileHandler, final Photo photo,
-        new MediaResolution[]{MediaResolution.LARGE, MediaResolution.PREVIEW,
-            MediaResolution.MEDIUM, MediaResolution.SMALL, MediaResolution.TINY};
-    BufferedImage previewImage = null;
-    for (MediaResolution mediaResolution : mediaResolutions) {
-      // Current thumbnail
-      HandledFile currentThumbnail = originalHandedImageFile.getParentHandledFile()
-          .getHandledFile(photoId + mediaResolution.getThumbnailSuffix());
-      // Thumbnail creation
-      resizePhoto((previewImage != null ? previewImage : originalImage), currentThumbnail,
-          (originalImageWidth > mediaResolution.getWidth() ? mediaResolution.getWidth() :
-              originalImageWidth), (watermark && mediaResolution.isWatermarkApplicable()),
-          nameWatermark,
-          (mediaResolution.getWatermarkSize() != null ? mediaResolution.getWatermarkSize() : 0));
-      // The first thumbnail that has to be created must be the larger one and without watermark.
-      // This first thumbnail is cached and reused for the following thumbnail creation.
-      if (previewImage == null) {
-        previewImage = ImageLoader.loadImage(currentThumbnail.getFile());
-      }
-    }
+  private static ImageTool getImageTool() {
+    return ImageTool.get();
   }
 
-  public static Size getWidthAndHeight(final String instanceId, final String subDir,
-      final String imageName, final int baseWidth) throws IOException {
-    return ImageUtility.getWidthAndHeight(instanceId, subDir, imageName, baseWidth);
+  private MediaUtil() {
   }
 
   /**
-   * Return the written file
-   * @param image the image from which the resolution must be red
-   * @param outputFile the output file
-   * @param widthParam the width parameter
-   * @param watermark the height parameter
-   * @param nameWatermark the watermark to write on the output file
-   * @param sizeWatermark the size of the watermark
-   * @throws Exception
+   * In charge of processing an internal media.
+   * @param <M>
    */
-  private static void resizePhoto(final BufferedImage image, final HandledFile outputFile,
-      final int widthParam, final boolean watermark, final String nameWatermark,
-      final int sizeWatermark) throws Exception {
-    OutputStream outputStream = null;
-    try {
-      outputStream = outputFile.openOutputStream();
-      final ImageResizer resizer = new ImageResizer(image, widthParam);
-      if (watermark) {
-        resizer.resizeImageWithWatermark(outputStream, nameWatermark, sizeWatermark);
-      } else {
-        resizer.resizeImage(outputStream);
-      }
-    } finally {
-      if (outputStream != null) {
-        IOUtils.closeQuietly(outputStream);
-      }
-    }
-  }
-
-  private static void createWatermark(final OutputStream watermarkedTargetStream,
-      final String watermarkLabel, final BufferedImage image, final int percentSizeWatermark)
-      throws IOException {
+  private static abstract class MediaProcess<M extends InternalMedia> {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Reorder the modifiers to comply with the Java Language Specification. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AModifiersOrderCheck)